File: src/main/java/com/mintos/weather/repository/IPRegionRepository.java
package com.mintos.weather.repository;

import com.mintos.weather.entity.IPRegionEntity;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Mono;

import java.util.UUID;

public interface IPRegionRepository extends ReactiveCrudRepository<IPRegionEntity, UUID> {

    Mono<IPRegionEntity> findByIp(String ip);

}

File: src/main/kotlin/com/mintos/weather/repository/IPRegionRepository.kt
package com.mintos.weather.repository

import com.mintos.weather.entity.IPRegionEntity
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import java.util.UUID

interface IPRegionRepository : CoroutineCrudRepository<IPRegionEntity, UUID> {

    suspend fun findByIp(ip: String): IPRegionEntity?
}